### PR TITLE
[Go SDK] Decrease sampling frequency for streaming jobs to avoid oversampling

### PR DESCRIPTION
**Pull request by zechenj18**
Decrease sampling frequency for streaming jobs to avoid oversampling

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by zechenj18** [comment]
R: @lostluck

**Comment by zechenj18** [comment]
R: @rohdesamuel

File: sdks/go/pkg/beam/core/runtime/exec/pcollection.go [hunk 1]
```diff
@@ -85,8 +86,8 @@ func (w *byteCounter) Write(p []byte) (n int, err error) {
 
 // ProcessElement increments the element count and sometimes takes size samples of the elements.
 func (p *PCollection) ProcessElement(ctx context.Context, elm *FullValue, values ...ReStream) error {
-	cur := atomic.AddInt64(&p.elementCount, 1)
-	if cur == p.nextSampleIdx {
+	cur := atomic.AddInt64(&p.bundleElementCount, 1)
+	if cur+p.pCollectionElementCount == p.nextSampleIdx {
```
Comment (lostluck) [inline sdks/go/pkg/beam/core/runtime/exec/pcollection.go:90]: Consider adding a comment here and/or where p.pCollectionElementCount is defined about the goal of this variable, and why it's safe to do this (local use only, instances aren't shared etc). I'm pretty confident in this change as is, but lets make it obvious to the next reader.

Comment (zechenj18) [reply sdks/go/pkg/beam/core/runtime/exec/pcollection.go:90]: Done: ) 
Added in pCollection struct definition.

**Review by lostluck — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**